spec/models/membership_spec.rb

```diff
@@ -0,0 +1,19 @@
+# frozen_string_literal: true
+
+require 'rails_helper'
+
+RSpec.describe Membership, type: :model do
+  let(:membership) { create(:membership) }
+
+  describe '#mark_as_revoked' do
+    it 'revokes the membership with a Time' do
+      expect(membership.status).to eq('active')
+      expect(membership.revoked_at).to eq(nil)
+
+      membership.mark_as_revoked!
+
+      expect(membership.status).to eq('revoked')
+      expect(membership.revoked_at).to be_a(Time)
```

Comment:
What about writing it like this?

```suggestion
      freeze_time do
        expect { membership.mark_as_revoked! }
          .to change { membership.reload.status }.from(:active).to(:revoked)
          .and change(membership, :revoked_at).from(nil).to(Time.current)
      end
```

Reply:
Od didn't knew this notation, gonna use it yeah !

My only concern is with `to(Time.current)`. I tested it before but there is a small diff between the value in test and the value stored in the `revoked_at` attribute, hence my check on `.to be_a(Time)`

Do you know an other way to do or is the `to be_a(Time)` check ok ?

Reply:
That's why I used `freeze_time` (https://apidock.com/rails/v6.1.3.1/ActiveSupport/Testing/TimeHelpers/freeze_time) ;)

Reply:
ooohhh ok nice 🙏

Reply:
Updated